core/src/main/scala/cilib/Fitness.scala

```diff
@@ -88,82 +81,88 @@ object Comparison {
-        case (Multi(xs), Multi(ys)) =>
+      (x.fitness, y.fitness) match {
+        case (-\/(f1), -\/(f2)) => fitCompare(f1, f2, x.violationCount, y.violationCount)
+        case (\/-(xs), \/-(ys)) =>
           val z = xs.zip(ys)
-          val x = z.forall {
+          val x2 = z.forall {
             case (a, b) =>
-              val r = fitCompare(a.f, b.f)
+              val r = fitCompare(a, b, x.violationCount, y.violationCount)
               r == LT || r == EQ
           }
-          val y = z.exists { case (a, b) => fitCompare(a.f, b.f) == LT }
+          val y2 = z.exists {
+            case (a, b) => fitCompare(a, b, x.violationCount, y.violationCount) == LT
+          }
 
-          if (x && y) LT else if (x) EQ else GT
+          if (x2 && y2) LT else if (x2) EQ else GT
 
-        case _ => sys.error("Cannot compare multiple objective against a single objective")
+        case _ => EQ
       }
   }
 }
 
 final case object Max extends Opt {
   private val D = Order[Double]
-  protected def fitCompare(x: Fit, y: Fit) =
+  private val I = Order[Int]
+
+  protected def fitCompare(x: Fit, y: Fit, xv: => Int, yv: => Int) =
     (x, y) match {
-      case (Adjusted(_, a), Adjusted(_, b))   => D.order(a, b)
-      case (Adjusted(_, a), Feasible(b))      => D.order(a, b)
-      case (Adjusted(_, _), Infeasible(_, _)) => GT
-      case (Feasible(a), Adjusted(_, b))      => D.order(a, b)
-      case (Feasible(a), Feasible(b))         => D.order(a, b)
-      case (Feasible(_), Infeasible(_, _))    => GT
-      case (Infeasible(_, _), Adjusted(_, _)) => LT
-      case (Infeasible(_, _), Feasible(_))    => LT
-      case (Infeasible(_, as), Infeasible(_, bs)) =>
-        if (as < bs) GT else if (as > bs) LT else EQ
+      case (Adjusted(_, a), Adjusted(_, b)) => D.order(a, b)
+      case (Adjusted(_, a), Feasible(b))    => D.order(a, b)
+      case (Adjusted(_, _), Infeasible(_))  => GT
+      case (Feasible(a), Adjusted(_, b))    => D.order(a, b)
+      case (Feasible(a), Feasible(b))       => D.order(a, b)
+      case (Feasible(_), Infeasible(_))     => GT
+      case (Infeasible(_), Adjusted(_, _))  => LT
+      case (Infeasible(_), Feasible(_))     => LT
+      case (Infeasible(a), Infeasible(b)) =>
+        if (xv == yv) D.order(a, b)
+        else I.order(xv, yv)
     }
 
   def objectiveOrder[A] = new Order[Objective[A]] {
     def order(x: Objective[A], y: Objective[A]) =
-      (x, y) match {
-        case (Single(f1, _), Single(f2, _)) => fitCompare(f1, f2)
-        case (Multi(xs), Multi(ys)) =>
+      (x.fitness, y.fitness) match {
+        case (-\/(f1), -\/(f2)) => fitCompare(f1, f2, x.violationCount, y.violationCount)
+        case (\/-(xs), \/-(ys)) =>
           val z = xs.zip(ys)
-          val x = z.forall {
+          val x2 = z.forall {
             case (a, b) =>
-              val r = fitCompare(a.f, b.f)
+              val r = fitCompare(a, b, x.violationCount, y.violationCount)
               r == GT || r == EQ
```

Comment:
I don't understand why we're checking for GT here. Shouldn't it be the same as for min and the only difference is the ordering we use for the fitness? Even the fitCompare function can be the same one as for Min

Reply:
I'll re-look at the logic - it's been a while and the changes were made to simply make the code compile.